examples/storage/README.md

```diff
@@ -2,7 +2,14 @@
 
 ## About
 
-This application demonstrates use of the ariel-os-storage API.
+This application demonstrates use of the
+[ariel-os-storage](https://ariel-os.github.io/ariel-os/dev/docs/api/ariel_os/storage/index.html)
+API.
+
+The example writes a number of key-value pairs to storage and retrieves them.
+Different types of values are written to storage to demonstrate the capablities.
+Values are also retrieved with different types as they were stored with, to show
+how values are decoded when different types are used between `insert` and `get`.
```

Comment:
Should this contain a note to Not Do This At Home?

Reply:
Reiterated the message from the storage module docs